internal/controller/pkg/revision/reconciler.go

```diff
@@ -386,22 +391,87 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 	// TODO(negz): Use Unhealthy().WithMessage(...) to supply error context?
 
-	// Initialize parser backend to obtain package contents.
-	reader, err := r.backend.Init(ctx, PackageRevision(pr))
-	if err != nil {
-		pr.SetConditions(v1.Unhealthy())
-		_ = r.client.Status().Update(ctx, pr)
+	pullPolicyNever := false
+	id := pr.GetName()
+	// If packagePullPolicy is Never, the identifier is the package source and
+	// contents must be in the cache.
+	if pr.GetPackagePullPolicy() != nil && *pr.GetPackagePullPolicy() == corev1.PullNever {
+		pullPolicyNever = true
+		id = pr.GetSource()
+	}
+
+	var rc io.ReadCloser
+	var err error
```

Comment:
Could this `err` be more tightly scoped? I think it's only used in the `if r.cache.Has` block, right?

Reply:
Absolutely, I think this is a relic of a previous iteration, thanks for pointing out!